llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -589,22 +589,36 @@ void StackMaps::recordStackMapOpers(
 
   const TargetRegisterInfo *TRI = AP.MF->getSubtarget().getRegisterInfo();
   std::set<int64_t> TrackedRegisters;
+
+  // Patchpoints and stackmaps have different instruction formats.
+  // Patchpoints embed function call arguments directly into the
+  // instruction, while stackmaps are simpler and only contain the
+  // stackmap data - therefore we need to treat them differently.
```

Comment:
This doesn't say HOW we have to treat them differently, i.e. that the tracked operands appear later in the instruction and there's some stuff we need to skip first. With the addition of the `PatchpointOpers` from the previous commit I don't think we need the lenghty examples of how patchpoints and stackmaps look like. The explanation and the use of `PatchpointOpers` should make this clear.

I'm not also wondering if there's a `StackmapOpers` we could also use instead of just blindly tracking all operands of a stackmap instruction?

Reply:
updated :point_right:  4fd45ba6f1ad1a842ffce468870fc3065a3c3f5f